telesto/apps/segmentation.py

```diff
@@ -0,0 +1,158 @@
+import base64
+import io
+import os
+import socket
+import threading
+import time
+from importlib import import_module
+from pathlib import Path
+from typing import Type, Deque
+from uuid import uuid4
+import shutil
+import json
+from collections import deque
+
+from PIL import Image
+import falcon
+
+from telesto.models.segmentation import DummySegmentationModel, SegmentationModelBase
+from telesto.logger import logger
+
+
+def preprocess(doc: dict) -> Image:
+    image_bytes = base64.b64decode(doc["image"]["content"])
+    image = Image.open(io.BytesIO(image_bytes))
+    return image
+
+
+def image_to_base64(image: Image) -> bytes:
+    fp = io.BytesIO()
+    image.save(fp, format="PNG")
+    fp.seek(0)
+    return base64.b64encode(fp.read())
+
+
+def postprocess(image: Image) -> dict:
+    image_b64 = image_to_base64(image)
+    return {"mask": {"content": image_b64.decode()}}
+
+
+class SegmentationBase:
+    def on_get(self, req, resp):
+        doc = {"status": "ok", "host": socket.getfqdn(), "worker.pid": os.getpid()}
+        resp.body = json.dumps(doc, ensure_ascii=False)
+
+
+class ImageStorage:
+
+    def __init__(self, base_path: str = "/tmp/telesto/images"):
+        self._base_path = Path(base_path)
+        self._base_path.mkdir(parents=True, exist_ok=True)
+
+    def clean(self):
+        shutil.rmtree(self._base_path, ignore_errors=True)
+        self._base_path.mkdir(parents=True, exist_ok=True)
+
+    def _image_path(self, gid: str, is_mask: bool) -> Path:
+        image_type = "mask" if is_mask else "image"
+        return self._base_path / f"{gid}-{image_type}.png"
+
+    def save(self, gid: str, image: Image, is_mask: bool):
+        image.save(self._image_path(gid, is_mask), format="PNG")
+
+    def load(self, gid: str, is_mask: bool) -> Image:
+        image_path = self._image_path(gid, is_mask)
+        if image_path.exists():
+            return Image.open(image_path)
+
+
+class SegmentationJobs:
```

Comment:
I am a bit worried that this part of the code will be in the public repository, but I remember this was the plan in the drawing.